src/core/cgroup.c

```diff
@@ -3058,11 +3058,28 @@ int unit_check_oomd_kill(Unit *u) {
         if (!increased)
                 return 0;
 
-        if (n > 0)
+        n = 0;
+        value = mfree(value);
+        r = cg_get_xattr_malloc(SYSTEMD_CGROUP_CONTROLLER, u->cgroup_path, "user.oomd_kill", &value);
+        if (r >= 0 && !isempty(value)) {
+                r = safe_atou64(value, &n);
+                if (r < 0)
+                        n = 0;
```

Comment:
you can drop the `r < 0` check and do
```
(void) safe_atou64(value, &n);
```

the convention in the systemd code base (including this function) is to only set `ret` values passed in on success.